ml_service/util/env_variables.py

```diff
@@ -21,32 +22,105 @@ class Env:
     run_evaluation: Optional[str] = os.environ.get("RUN_EVALUATION", "true")
-    allow_run_cancel: Optional[str] = os.environ.get("ALLOW_RUN_CANCEL",
-                                                     "true")
+    allow_run_cancel: Optional[str] = os.environ.get(
+        "ALLOW_RUN_CANCEL", "true"
+    )  # NOQA: E501
     aml_env_name: Optional[str] = os.environ.get("AML_ENV_NAME")
+    aml_env_train_conda_dep_file: Optional[str] = os.environ.get(
+        "AML_ENV_TRAIN_CONDA_DEP_FILE", "conda_dependencies.yml"
+    )
     rebuild_env: Optional[bool] = os.environ.get(
-        "AML_REBUILD_ENVIRONMENT", "false").lower().strip() == "true"
+        "AML_REBUILD_ENVIRONMENT", "false"
+    ).lower().strip() == "true"
+
+    use_gpu_for_scoring: Optional[bool] = os.environ.get(
+        "USE_GPU_FOR_SCORING", "false"
+    ).lower().strip() == "true"
+    aml_env_score_conda_dep_file: Optional[str] = os.environ.get(
+        "AML_ENV_SCORE_CONDA_DEP_FILE", "conda_dependencies_scoring.yml"
+    )
+    aml_env_scorecopy_conda_dep_file: Optional[str] = os.environ.get(
+        "AML_ENV_SCORECOPY_CONDA_DEP_FILE", "conda_dependencies_scorecopy.yml"
+    )
+    vm_size_scoring: Optional[str] = os.environ.get(
+        "AML_COMPUTE_CLUSTER_CPU_SKU_SCORING"
+    )
+    compute_name_scoring: Optional[str] = os.environ.get(
+        "AML_COMPUTE_CLUSTER_NAME_SCORING"
+    )
+    vm_priority_scoring: Optional[str] = os.environ.get(
+        "AML_CLUSTER_PRIORITY_SCORING", "lowpriority"
+    )
+    min_nodes_scoring: int = int(
+        os.environ.get("AML_CLUSTER_MIN_NODES_SCORING", 0)
+    )  # NOQA: E501
+    max_nodes_scoring: int = int(
+        os.environ.get("AML_CLUSTER_MAX_NODES_SCORING", 4)
+    )  # NOQA: E501
+    rebuild_env_scoring: Optional[bool] = os.environ.get(
+        "AML_REBUILD_ENVIRONMENT_SCORING", "false"
+    ).lower().strip() == "true"
+    scoring_datastore_storage_name: Optional[str] = os.environ.get(
+        "SCORING_DATASTORE_STORAGE_NAME"
+    )
+    scoring_datastore_access_key: Optional[str] = os.environ.get(
+        "SCORING_DATASTORE_ACCESS_KEY"
+    )
+    scoring_datastore_input_container: Optional[str] = os.environ.get(
+        "SCORING_DATASTORE_INPUT_CONTAINER"
+    )
+    scoring_datastore_input_filename: Optional[str] = os.environ.get(
+        "SCORING_DATASTORE_INPUT_FILENAME"
+    )
+    scoring_datastore_output_container: Optional[str] = os.environ.get(
+        "SCORING_DATASTORE_OUTPUT_CONTAINER"
+    )
+    scoring_datastore_output_filename: Optional[str] = os.environ.get(
+        "SCORING_DATASTORE_OUTPUT_FILENAME"
+    )
+    scoring_dataset_name: Optional[str] = os.environ.get(
+        "SCORING_DATASET_NAME"
+    )  # NOQA: E501
+    scoring_pipeline_name: Optional[str] = os.environ.get(
+        "SCORING_PIPELINE_NAME"
+    )  # NOQA: E501
+    aml_env_name_scoring: Optional[str] = os.environ.get(
+        "AML_ENV_NAME_SCORING"
```

Comment:
Could the two environments, conda dependency files etc be combined together so there is just one `scoring` environment instead of having the `scoring` and `score_copy` environment? It would only add one additional package dependency to the scoring environment I believe.

Reply:
We could, but I am not sure it improves anything. IIRC each pipeline step creates its own containers, and since the scoring step and the copying step have different package requirements, keeping them separate in terms of dependencies makes sense to me.